test/distributed/_sharded_tensor/test_sharded_tensor.py

```diff
@@ -0,0 +1,236 @@
+import torch
+import torch.distributed as dist
+from torch.distributed import _sharded_tensor
+from torch.distributed._sharding_spec import ChunkShardingSpec
+from torch.testing._internal.common_distributed import (
+    MultiProcessTestCase,
+    requires_nccl,
+    skip_if_lt_x_gpu,
+)
+
+
+class TestShardedTensor(MultiProcessTestCase):
+    @property
+    def world_size(self):
+        return 4
+
+    def init_pg(self):
+        dist.init_process_group(
+            backend="nccl",
+            world_size=self.world_size,
+            rank=self.rank,
+            init_method=f"file://{self.file_name}",
+        )
+
+    def setUp(self) -> None:
+        super().setUp()
+        self._spawn_processes()
+
+    @skip_if_lt_x_gpu(4)
+    @requires_nccl()
+    def test_chunked_sharding_basic(self):
```

Comment:
Nit: "basic" does not sound like a good descriptive test name. What about "shared_by_alll_4_devices"?